python/tchannel/tornado/connection.py

```diff
@@ -456,3 +450,87 @@ def ping(self):
 
     def pong(self):
         return self._write(messages.PingResponseMessage())
+
+
+class StreamConnection(TornadoConnection):
+    def __init__(self, connection):
+        super(StreamConnection, self).__init__(connection)
+
+    @tornado.gen.coroutine
+    def _stream(self, context, message_factory):
+        """write request/response into frames
+
+        Transform request/response into protocol level message objects based on
+        types and argstreams.
+
+        Assumption: the chunk data read from stream can fit into memory.
+
+        If arg stream is at init or streaming state, build the message based on
+        current chunk. If arg stream is at completed state, put current chunk
+        into args[] array, and continue to read next arg stream in order to
+        compose a larger message instead of sending multi small messages.
+
+        Note: the message built at this stage is not guaranteed the size is
+        less then 64KB.
+
+        Possible messages created sequence:
+
+        Take request as an example:
+        CallRequestMessage(flags=fragment)
+            --> CallRequestContinueMessage(flags=fragment)
+            ....
+            --> CallRequestContinueMessage(flags=fragment)
+                --> CallRequestMessage(flags=none)
+
+        :param context: Request or Response object
+        """
+        args = []
+        for i, argstream in enumerate(context.argstreams):
+            chunk = yield argstream.read()
+            args.append(chunk)
+            chunk = yield argstream.read()
+            while chunk:
+                message = (message_factory.
+                           build_raw_message(context, args))
+                self.write(message, context.id)
+                args = [chunk]
+                chunk = yield argstream.read()
+
+        # last piece of request/response.
+        message = (message_factory.
+                   build_raw_message(context, args, is_completed=True))
+        self.write(message, context.id)
+
+    @tornado.gen.coroutine
+    def post_response(self, response):
+        response.close_argstreams()
+        yield self._stream(response, self.response_message_factory)
+        response.close_argstreams(force=True)
+
+    @tornado.gen.coroutine
+    def post_request(self, request):
+        """send the given request and response is not required"""
+        request.close_argstreams()
+        yield self._stream(request, self.request_message_factory)
+        request.close_argstreams(force=True)
+
+    def send_request(self, request):
+        """Send the given request and response is required.
+        Use this for messages which have a response message.
```

Comment:
Add empty lines before and after "Use this for messages which have a response message"